### PR TITLE
chef_ingredient as a custom resource

### PR DESCRIPTION
**Pull request by wrightp**
### Description
- Convert ChefIngredient LWRP into a custom resource
- Remove chef_server_ingredient shims
- Updated ingredient_config and omnibus_service resources

### Check List

- [x] rebase and sqush
- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

@chef-cookbooks/engineering-services

### CODE REVIEW

File: resources/chef_ingredient.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+# platforms that are not yet officially added to Chef support matrix
+# TODO: *this should not be defaulted. This should be handled by mixlib-install by default*
+property :platform_version_compatibility_mode, kind_of: [TrueClass, FalseClass], default: false
+
+# Configure specific platform package
+property :platform, kind_of: String
+property :platform_version, kind_of: String
+property :architecture, kind_of: String
+
+platform_family = node['platform_family']
```
Comment (schisamo) [inline resources/chef_ingredient.rb]: Why can't the `node['platform_family']` be placed down in the `action_class` block?

Comment (wrightp) [inline resources/chef_ingredient.rb]: node is not available within the block.

**Comment by wrightp** [comment]
Hmm. Looking into these travis failures. Weird

**Comment by wrightp** [comment]
Getting all this to work as a custom resource was quite simple. Getting to mocks to work however was probably more challenging than it needed to be. Thank you spy.

@schisamo 

I'm would like to follow up this PR with a few more changes.
- update to latest mixlib-install 3.x (to be released)
- change platform_compat behavior
- Determine whether we can safely remove the omnitruck handler

**Comment by rhass** [comment]
I think we should add a minor version bump to the metadata to insure it is done for the release.

**Comment by wrightp** [comment]
@rhass This PR has breaking changes. Once this is released it will require a major bump.

**Pull request merged**